Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1625,6 +1625,303 @@ function Test-DiscoveryService{
+
+    foreach($nodeToSearch in $nodesToSearch)
+    {
+     $setAzureSettings = $azureIdentityScope.SelectSingleNode($nodeToSearch)
+
+     if($reverseOperator -eq "not equal")
+     {
+       $removeEmptyVariables = $setAzureSettings.ChildNodes | Where-Object {$_.$childNodeGetAttribute -ne $childNodeAttributeSetting}
+     }
+     else
+     {
+       $removeEmptyVariables = $setAzureSettings.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+     }
+
+     if($null -ne $removeEmptyVariables)
+     {
+      foreach($removeEmptyVariable in $setAzureSettings.variable){
+       $removeEmptyVariable.ParentNode.RemoveChild($removeEmptyVariable)
+      }
+     }
+    }
+    $azureInstallationConfig.Save("$azureConfigPath") | Out-Null
+}
+
+function Get-XMLChildNode {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $installConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeSetting
+    )
+
+    $installationConfig = [xml](Get-Content $installConfigPath)
+    $identityScope = $installationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+    $existingChildNode = @{}
+
+    $existingChildNode = $identityScope.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+    if ($null -eq $existingChildNode)
+    {
+      Write-DosMessage -Level "Information" -Message "$childNodeAttributeSetting not found in install.config"
+    }
+    return $existingChildNode
+}
+
+function Set-XMLChildNode {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $azureConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $configSection,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeGetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeSetAttribute,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeSetting,
+        [Parameter(Mandatory=$true)]
+        [string] $childNodeAttributeValue
+    )
+
+    $azureInstallationConfig = [xml](Get-Content $azureConfigPath)
+    $azureIdentityScope = $azureInstallationConfig.installation.settings.scope | Where-Object {$_.name -eq $configSection}
+    $existingChildNode = @{}
+
+    $existingChildNode = $azureIdentityScope.ChildNodes | Where-Object {$_.$childNodeGetAttribute -eq $childNodeAttributeSetting}
+    if ($null -ne $existingChildNode)
+    {
+      $existingChildNode.$childNodeSetAttribute = $childNodeAttributeValue
+      $azureInstallationConfig.Save("$azureConfigPath") | Out-Null
+      Write-DosMessage -Level "Information" -Message "$childNodeAttributeValue added to $childNodeAttributeSetting in azureSettings.config"
+    }
+    elseif ($null -eq $existingChildNode)
```

Comment:
At this point this is just setting values right? So if the node doesn't have a value, then it just stays as empty right?

Reply:
Since the parameter is mandatory, $childNodeAttributeValue would have to be passed, not empty string, for the function to not fail.